boards: add Wemos D1 R32 (aka ESPDuino-32) board definition (#21479)
RIOT-OS/RIOT: boards/esp32-wemos-d1-r32/doc.md

The Wemos D1 R32 board is an ESP32 board in the Arduino Uno format that uses the
ESP32-WROOM-32 module. It is exactly the same board that is also known as
ESPDuino-32 on the market. The board is pin compatible with the Arduino Uno
board. The peripheral configuration corresponds to the Arduino Uno pinout and
guaranties the compatibility with Arduino Uno Shields.

\image html "https://makerselectronics.com/wp-content/uploads/2022/09/wemos-d1-r32-w-esp32-uno-r3-pinout-1-800x800.jpg.webp" "WeMos D1 R32 ESPDuino32" width=400px

[Back to table of contents](#esp32_wemos_d1_r32_toc)

## Hardware {#esp32_wemos_d1_r32_hardware}

This section describes

- the [MCU](#esp32_wemos_d1_r32_mcu),
- the [board configuration](#esp32_wemos_d1_r32_board_configuration),
- the [board pinout](#esp32_wemos_d1_r32_pinout).

[Back to table of contents](#esp32_wemos_d1_r32_toc)

### MCU {#esp32_wemos_d1_r32_mcu}

Most features of the board are provided by the ESP32 SoC. For detailed
information about the ESP32, see section \ref esp32_mcu_esp32 "MCU ESP32".

[Back to table of contents](#esp32_wemos_d1_r32_toc)

### Board Configuration {#esp32_wemos_d1_r32_board_configuration}

> The boad configuration guaranties that the board is compatible with Arduino

Reviewer: ```suggestion
The board configuration guarantees that the board is compatible with Arduino
```

Author: Good catch, I'm wondering why it wasn't catched by the spell checker in CI?

Reviewer: `guaranties` is the valid plural form of `guaranty` and `boad` is simply not in the library: https://github.com/codespell-project/codespell/blob/d59cf4fe68f1d606c23f21a19ecd3828dbe8725e/codespell_lib/data/dictionary.txt#L9789

> /*
>  * Copyright (C) 2025 Gunar Schorcht
>  *
>  * This file is subject to the terms and conditions of the GNU Lesser
>  * General Public License v2.1. See the file LICENSE in the top level
>  * directory for more details.
>  */
> 
> /**

Reviewer: ```suggestion
```

No copyright notices and block comments are required for Markdown files.
(Remember to remove the closing piece of the block comment below as well).

>  * @defgroup    boards_esp32_wemos_d1_r32 Wemos D1 R32 (ESPDuino-32)
>  * @ingroup     boards_esp32
>  * @brief       Support for the Wemos D1 R32 (ESPDuino-32) board
>  * @author      Gunar Schorcht <gunar@schorcht.net>

Reviewer: ```suggestion
@defgroup    boards_esp32_wemos_d1_r32 Wemos D1 R32 (ESPDuino-32)
@ingroup     boards_esp32
@brief       Support for the Wemos D1 R32 (ESPDuino-32) board
@author      Gunar Schorcht <gunar@schorcht.net>
```

Author: Ah ok, now where it is a Markdown file, C comments are wrong now :sunglasses:

\section esp32_wemos_d1_r32 Wemos D1 R32 board (ESPDuino-32)

## Table of Contents {#esp32_wemos_d1_r32_toc}

> 1. [Overview](#esp32_wemos_d1_r32_overview)
> 2. [Hardware](#esp32_wemos_d1_r32_hardware)
>     1. [MCU](#esp32_wemos_d1_r32_mcu)
>     2. [Board Configuration](#esp32_wemos_d1_r32_board_configuration)
>     3. [Board Pinout](#esp32_wemos_d1_r32_pinout)
> 3. [Flashing the Device](#esp32_wemos_d1_r32_flashing)

Reviewer: ```suggestion
-# [Overview](#esp32_wemos_d1_r32_overview)
-# [Hardware](#esp32_wemos_d1_r32_hardware)
    -# [MCU](#esp32_wemos_d1_r32_mcu)
    -# [Board Configuration](#esp32_wemos_d1_r32_board_configuration)
    -# [Board Pinout](#esp32_wemos_d1_r32_pinout)
-# [Flashing the Device](#esp32_wemos_d1_r32_flashing)
```

You can let Doxygen do the enumeration for you instead of doing it manually.
